addons/io/org.openhab.io.transport.modbus.test/src/test/java/org/openhab/io/transport/modbus/test/SmokeTest.java

```diff
@@ -746,27 +731,32 @@ public void onBits(ModbusReadRequestBlueprint request, BitArray bits) {
         modbusManager.registerRegularPoll(task, 150, 0);
         callbackCalled.await(5, TimeUnit.SECONDS);
         modbusManager.unregisterRegularPoll(task);
-        long countAfter = callbackCalled.getCount();
-
-        //
-        // Test poll rate
-        //
         long end = System.currentTimeMillis();
-        int responses = dataReceived.get();
-        assertTrue(responses > 1);
+        assertPollDetails(unexpectedCount, dataReceived, start, end, 145, 500);
+    }
 
+    /**
+     *
+     * @param unexpectedCount number of unexpected callback calls
+     * @param callbackCalled number of callback calls (including unexpected)
+     * @param dataReceived number of expected callback calls (onBits or onRegisters)
+     * @param pollStartMillis poll start time in milliepoch
+     * @param expectedPollAverageMin average poll period should be at least greater than this
+     * @param expectedPollAverageMax average poll period less than this
+     * @throws InterruptedException
+     */
+    private void assertPollDetails(AtomicInteger unexpectedCount, AtomicInteger expectedCount, long pollStartMillis,
+            long pollEndMillis, int expectedPollAverageMin, int expectedPollAverageMax) throws InterruptedException {
+        int responses = expectedCount.get();
         assertThat(unexpectedCount.get(), is(equalTo(0)));
```

Comment:
This is not an OSGi test hence lets use `assertEquals()` here.

Reply:
I don't understand this comment, is there rule against using hamcrest in non-OSGi tests?

Reply:
Um, ... Maybe I misunderstood the latest discussions about  it.. Both are fine - but  Hamcrest preferred over JUnit in OSGi tests. Sry for confusion.

https://github.com/eclipse/smarthome/blob/master/docs/documentation/development/testing.md#unit-tests

Let's keep it like it is.